### DISCUSSION TITLE
Ability to write arbitrary messages

### DISCUSSION THREAD

**Discussion by wrongisthenewright** [john30/ebusd#810] [Q&A]
Hi,
I'm using ebusd to communicate with an Ariston hybrid system, Ariston group systems use ebus with a proprietary and undocumented protocol, most ebus logic are respected so reading is almost straightforward. writing command on the bus is instead more complicated as the syntax is different. 
I'm able to send my "custom" command on the bus using the hex function, but that is incompatible with mqtt topics. 

It'd  be great if in some way one could define a message structure in a config file and pass via mqtt the variable payload to ebusd. ebusd could get the value from mqtt, translate it in ebus protocol (i.e DEC-->HEX-->EbusDataType) and then write it on the bus.

Any idea on how to do it?

**Comment by ysard** [comment]
Hi, the best is to create one rule (one for read, and one for write with the same name) for every settings you want to use on MQTT.

The file mqtt-hassio.cfg is not a straightforward format, but you have mainly to understand that it's based on regex/pattern detection.

Our rules for Ariston/Chaffoteaux are slightly different from the rules written for Vaillant for which mqtt-hassio.cfg was made.
My rules are also slightly different than yours (some renaming), but my config file should be compatible.

See the readme, and the commits about it, and do not hesitate to create an issue if explanations are missing:
https://github.com/ysard/ebusd_configuration_chaffoteaux_bridgenet/
https://github.com/ysard/ebusd_configuration_chaffoteaux_bridgenet/search?q=%5Bebusd_HA%5D&type=commits

**Reply by wrongisthenewright** [reply]
> Hi, the best is to create one rule (one for read, and one for write with the same name) for every settings you want to use on MQTT.
>
Hi,
that's an interesting solution, I was thinking (wronlgly) that 2 CSV lines couldn't share the same CIRCUIT,NAME,COMMENT,QQ,ZZ,PBSB,ID
sequence, but now I discovered that I can do it if the TYPE is different (read vs. write), awesome!
 
> The file mqtt-hassio.cfg is not a straightforward format, but you have mainly to understand that it's based on regex/pattern detection.

OK, I hate regexp, but I'll have to cope.... I'll look into your files and add the metrics/command/sensors that my system needs (the hybrid and heatpump codes missing).


> See the readme, and the commits about it, and do not hesitate to create an issue if explanations are missing: https://github.com/ysard/ebusd_configuration_chaffoteaux_bridgenet/ https://github.com/ysard/ebusd_configuration_chaffoteaux_bridgenet/search?q=%5Bebusd_HA%5D&type=commits

TY
